### ISSUE TITLE
Fails to display help when serial is unavailable

### ISSUE THREAD

**Issue by y2kbugger** [devantech/iceFUNprog#5]
Steps to reproduce:

Run iceFUNprog without a serial device connected.

**Comment by devantech** [comment]
iceFUNprog correctly issues an error message:
iceFUNprog: failed to open serial port.

**Comment by Sadoon-AlBader** [comment]
I think what @y2kbugger meant was that even when you run `iceFUNprog -h` or `iceFUNprog --help`, if the serial device is not connected or permissions are not set up correctly, it will not display the help menu.
That's because in the code iceFUNprog checks on the port before checking the command line arguments.
This can also be problematic if the user has two serial USB devices connected. I can't program the board while my arduino is attached for that reason, it just ignores the `-P` option and returns an error.

```
$ iceFUNprog -P /dev/ttyACM1 build/blinky.bin
iceFUNprog: failed to open serial port.
```

**Comment by devantech** [comment]
We have updated the code to display help when there is no serial connection present. Hopefully the change should resolve the issue.

**Comment by Sadoon-AlBader** [comment]
There was also a related issue where it ignores the supplied port with "-P", I have fixed it all in my pull request #13
